Userland/Libraries/LibThreading/ConditionVariable.h

```diff
@@ -34,6 +36,11 @@ class ConditionVariable {
     {
         pthread_cond_wait(&m_condition, &m_to_wait_on.m_mutex);
     }
+    ALWAYS_INLINE void wait_while(Function<bool()> condition)
+    {
+        while (condition())
+            wait();
+    }
```

Comment:
Shouldn't this go into the previous commit?

Reply:
Yes it should. 🤦‍♂️

Reply:
Also, the function in _Let's write synchronization primitives_ waits while the condition is _not_ true (so it's more of a "wait until" or "wait for" than "wait while"). I don't really have a preference either way, but keeping consistency with LWSP is a weak argument in favor of `wait_until()`.

What do y'all think?